src/main/bin/ccm-initialise

```diff
@@ -7,131 +7,121 @@
 # Initialise local config cache
 #
 
+package ccm_initialise;
+
+use strict;
+use warnings;
+
+# required for CAF (Common Application Framework)
 BEGIN {
-    # use perl libs in /usr/lib/perl
     unshift(@INC, '/usr/lib/perl');
 }
 
-use strict;
-use warnings;
-use Getopt::Long;
-use File::Path;
-use EDG::WP4::CCM::CCfg;
+use CAF::Application;
+use CAF::Reporter;
+use LC::Exception qw(SUCCESS throw_error);
+our @ISA = qw(CAF::Application CAF::Reporter);
 
-# Options
-my $debug = undef;
-my $config = undef;
-my $cache_root = undef;
-my $world_readable = 0;
-my $group_readable = undef;
+# needed for method overriding
+no warnings 'redefine';
 
-sub Quit
-{
-    my $msg = shift;
+#
+# application specific options:
+#
+sub app_options {
 
-    if ($msg) {
-        print STDERR "cache-initialise: $msg\n";
-        exit(1);
-    } else {
-        exit(0);
-    }
-}
+    # local lexicals
+    my @options;
 
-sub Warn
-{
-    my $msg = shift;
+    # build options
+    push (@options,
 
-    print "[WARN] $msg \n";
+          { NAME    => 'cfgfile=s',
+            HELP    => 'configuration file for ccm-initialise' },
 
-    return 'ok';
+          { NAME    => 'debug|d=i',
+            DEFAULT => '0',
+            HELP    => 'Turn on debugging messages' });
+
+    return \@options;
 }
 
-sub Debug
-{
-    my $msg = shift;
-    if (defined($debug)) {
-        print $msg . "\n";
+#
+# initialise
+#
+sub _initialize {
+
+    # local lexicals
+    my $self = shift;
+
+    # version and usage
+    $self->{'VERSION'} = "abc";
```

Comment:
oops. some testing code got pushed. it contained the maven template (but give perl errors when testing untemplated :smile: ). will fix